File: src/0014-longest-common-prefix/solution.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
  let ans = strs[0];

  const get = function(s, t) {
    let a = "";
    for(let i = 0; i < s.length && i < t.length; i++) {
      if (s[i] === t[i]) a += s[i];
      else break;
    }
    return a;
  }

  for(let i = 1; i < strs.length; i++) {
    ans = get(ans, strs[i]);
  }

  return ans;  
};

File: src/0014-longest-common-prefix/solution.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        result = ""
        for i, c in enumerate(strs[0]):
            for s in strs:
                if i == len(s) or s[i] != c:
                    return result
            result += c
        return result
